contrib/datadog/tracing/interceptor.go

```diff
@@ -0,0 +1,262 @@
+// The MIT License
+//
+// Copyright (c) 2021 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package tracing provides DataDog tracing utilities
+package tracing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+// InterceptorOptions are options provided to NewInterceptor
+type InterceptorOptions struct {
+	// DisableSignalTracing can be set to disable signal tracing.
+	DisableSignalTracing bool
+
+	// DisableQueryTracing can be set to disable query tracing.
+	DisableQueryTracing bool
```

Comment:
Nit: I would rather have a generic `TracerOptions` struct in common across all the tracing implementations since we're ultimately only passing these through to the generic Tracer in the end and don't act on them at all in Datadog specific implementation.

Or better yet, perhaps we could add the ability to customize tracer options from https://pkg.go.dev/go.temporal.io/sdk@v1.17.0/interceptor#NewTracingInterceptor and simply return defaults from the Datadog tracer's `Options` method.

This comment isn't a blocker though; the pattern here seems consistent with the existing tracers.

Reply:
I think that would assume a bit too much similarity amongst tracers (even if that does seem to be the case today). Granted that is the case now, but from a user-facing POV having DataDog-specific values in here can work too in the future when you need them.

Specifically for `DisableSignalTracing` and `DisableQueryTracing` they were added as a safety measure when that tracing was turned on by default. I'd be ok if y'all didn't want to expose these.

I would ask though, even if y'all get rid of every option, to have an empty struct for future proofing (we prefer the struct-as-config model over the options-type vararg pattern).